core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -269,8 +270,8 @@ object LogConfig {
    */
   def fromProps(defaults: java.util.Map[_ <: Object, _ <: Object], overrides: Properties): LogConfig = {
     val props = new Properties()
-    props.putAll(defaults)
-    props.putAll(overrides)
+    defaults.asScala.foreach { case (k, v) => props.put(k, v) }
```

Comment:
The key type is Object instead of String, which is unsafe. We should probably change the type of the map received by LogConfig but I didn't want to tackle that here.